mountpoint-s3-client/src/s3_crt_client/get_object.rs

```diff
@@ -99,14 +107,23 @@ pub struct S3GetObjectRequest {
     #[pin]
     finish_receiver: UnboundedReceiver<Result<GetBodyPart, Error>>,
     finished: bool,
+    enable_backpressure: bool,
+    next_offset: u64,
+    last_offset: Option<u64>,
+    next_read_window_offset: u64,
 }
 
 impl GetObjectRequest for S3GetObjectRequest {
     type ClientError = S3RequestError;
 
     fn increment_read_window(mut self: Pin<&mut Self>, len: usize) {
+        self.next_read_window_offset += len as u64;
         self.request.meta_request.increment_read_window(len as u64);
     }
+
+    fn next_read_window_offset(self: Pin<&Self>) -> u64 {
```

Comment:
This may require some comment or a better name. Why `next`? Isn't this the upper bound of the current read window?

Reply:
In my mind current offset is at the start of the read window (the lower bound), so the upper bound would be `next`. I will add the comment and also happy to change if it doesn't make sense.